### PR TITLE
Feat(eos_cli_config_gen): Support for "agents" config

### PR DESCRIPTION
**Pull request by gmuloc**
## Change Summary

Render stuff like:

```
agent BLAH environment BLAH=BLAH
agent KernelFib environment TOTO=TEAH
```

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
```
ceos2(config)#show cli commands | grep agent
[no|default] agent AGENT environment ENVIRON_CONFIG
[no|default] agent AGENT memory allocations heap fastbin limit SIZE
[no|default] agent AGENT shutdown [ supervisor ( active | standby | all ) ]
[no|default] agent AGENT_NAME warm-restart ( ( allowed ALLOWED_NUM ) | ( window WINDOW_NUM ) | always )
agent fatal-error action ACTION
```

```
ceos2(config)#agent BLAH environment ?
  EXPR  Agent environment configuration in the form of VAR1=VAL1:VAR2=VAL2:...
```

```yaml
    agents:
      - name: <str>
        environments:
          - name: <str>
            value: <str>
```

## How to test

molecule

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/agents.j2 [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+{% for agent in agents | arista.avd.natural_sort('name') %}
+{%     if loop.first %}
+!
+{%     endif %}
+{%     if agent.environments is arista.avd.defined %}
+{%         set env_list = [] %}
+{%         for envvar in agent.environments %}
+{%             do env_list.append(envvar.name ~ "=" ~ envvar.value) %}
+{%         endfor %}
+agent {{ agent.name }} environment  {{ env_list | join(":") }}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/agents.j2]:
```suggestion
agent {{ agent.name }} environment {{ env_list | join(":") }}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/agents.schema.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  agents:
+    type: list
+    primary_key: name
+    items:
+      type: dict
+      keys:
+        name:
+          type: str
+          description: Agent name.
+        environments:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/agents.schema.yml]: Just a suggestion for clarity, but I can accept the one you have as well.
```suggestion
        environment_variables:
```

Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/agents.schema.yml]: [6994314](https://github.com/aristanetworks/ansible-avd/pull/3282/commits/699431441517c470abe4c30ba61952e1bc6bd024)

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/agents.j2 (hunk 1 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/agents.j2]: [6994314](https://github.com/aristanetworks/ansible-avd/pull/3282/commits/699431441517c470abe4c30ba61952e1bc6bd024)

**Review by ClausHolbechArista — APPROVED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/agents.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<!--
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/agents.md:1]: Need to also add this to `ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/input-variables.md`

**Review by carlbuchmann — CHANGES_REQUESTED** [review]
Need to update eos_cli_config_gen input-variable.md

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/agents.md (hunk 3 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/agents.md:1]: [9065c77](https://github.com/aristanetworks/ansible-avd/pull/3282/commits/9065c77cf24a353da2e1bd0c65dab4277f1b226a)

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Pull request merged**